internal/contentcache/contentcache.go

```
@@ -127,27 +165,31 @@ func (c *ContentCache) NewTempFile(rc io.ReadCloser) (gcsx.TempFile, error) {
 }
 
 // AddOrReplace creates a new cache file or updates an existing cache file
-func (c *ContentCache) AddOrReplace(cacheObjectKey *CacheObjectKey, generation int64, rc io.ReadCloser) (gcsx.TempFile, error) {
+func (c *ContentCache) AddOrReplace(cacheObjectKey *CacheObjectKey, generation int64, rc io.ReadCloser) (*CacheObject, error) {
 	if cacheObject, exists := c.fileMap[*cacheObjectKey]; exists {
 		cacheObject.Destroy()
 	}
-	metadata := &gcsx.TempFileObjectMetadata{
-		BucketName: cacheObjectKey.BucketName,
-		ObjectName: cacheObjectKey.ObjectName,
-		Generation: generation,
+	file, err := c.NewCacheFile(rc)
```

This error is not handled